test/e2e/util/util.go

```diff
@@ -1167,6 +1195,434 @@ func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client,
 	})
 }
 
+// InstallCilium validates that Cilium network policies are properly enforced
+// in ARO HCP guest clusters. This test covers:Verifying Cilium installation
+func InstallCilium(t *testing.T, ctx context.Context, guestClient crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("InstallCilium", func(t *testing.T) {
+		if !azureutil.IsAroHCP() {
+			t.Skip("test only supported on ARO HCP clusters")
+		}
+
+		t.Run("InstallCiliumNameSpace", func(t *testing.T) {
+			g := NewWithT(t)
+			t.Log("Configuring cilium namespace with PodSecurity labels")
+			ciliumNs := &corev1.Namespace{}
+			err := guestClient.Get(ctx, crclient.ObjectKey{Name: "cilium"}, ciliumNs)
+
+			if apierrors.IsNotFound(err) {
+				// Namespace doesn't exist, create it with labels
+				t.Log("Creating cilium namespace")
+				ciliumNs = &corev1.Namespace{}
+				ciliumNs.Name = "cilium"
+				ciliumNs.Labels = map[string]string{
+					"security.openshift.io/scc.podSecurityLabelSync": "false",
+					"pod-security.kubernetes.io/enforce":             "privileged",
+					"pod-security.kubernetes.io/audit":               "privileged",
+					"pod-security.kubernetes.io/warn":                "privileged",
+				}
```

Comment:
Create a `manifest.CilliumNamespace` (or something similar) function with all the details to be called from here.